### PR TITLE
[lit-html] Fix cache() directive

### PR DESCRIPTION
**Pull request by justinfagnani**
The cache tests has a wayward `test.only` in them, hiding some failures. The directive wasn't clearing the container part at the right times, and was setting the container part's value wrong when restoring from cache. The fix should read a bit better now.

### CODE REVIEW

File: packages/lit-html/src/directives/cache.ts [hunk 1]
```diff
@@ -57,38 +57,42 @@ export const cache = directive(
     }
 
     update(containerPart: ChildPart, [v]: DirectiveParameters<this>) {
-      // If the new value is not a TemplateResult from the same Template as the
-      // previous value, move the nodes from the DOM into the cache.
+      // If the previous value is a TemplateResult and the new value is not,
+      // or is a different Template as the previous value, move the child part
+      // from into the cache.
```
Comment (kevinpschaaf) [inline packages/lit-html/src/directives/cache.ts:62]:
```suggestion
      // into the cache.
```

File: packages/lit-html/src/directives/cache.ts [hunk 2]
```diff
@@ -57,38 +57,42 @@ export const cache = directive(
...
           const fragment = new DocumentFragment();
           cachedContainerPart = render(nothing, fragment);
-          setComittedValue(cachedContainerPart, [childPart]);
           this.templateCache.set(this.value.strings, cachedContainerPart);
         }
         // Move into cache
-        insertPart(containerPart, undefined, childPart);
+        setComittedValue(cachedContainerPart, [childPart]);
+        insertPart(cachedContainerPart, undefined, childPart);
         clearPart(containerPart);
```
Comment (kevinpschaaf) [inline packages/lit-html/src/directives/cache.ts:78]: Should the clear here be necessary if we've definitely just moved the whole `childPart` which is what would have been committed there into the other part?

File: packages/lit-html/src/directives/cache.ts (hunk 1 above)
Comment (justinfagnani) [reply packages/lit-html/src/directives/cache.ts]: done

File: packages/lit-html/src/directives/cache.ts (hunk 2 above)
Comment (justinfagnani) [reply packages/lit-html/src/directives/cache.ts]: ah, good point

**Review by kevinpschaaf — APPROVED** [review]
LGTM!

**Pull request merged**